torch/_decomp/decompositions.py

```diff
@@ -1237,3 +1237,73 @@ def fast_pow(x, ord):
         self = self.abs()
 
     return fast_pow(fast_pow(self, p).sum(dim, keepdim=keepdim), 1.0 / p)
+
+
+@register_decomposition(torch.ops.aten.upsample_bilinear2d.vec)
+def upsample_bilinear2d_vec(
+    input: Tensor,
+    output_size: Optional[List[int]],
+    align_corners: bool,
+    scale_factors: Optional[List[float]],
+) -> Tensor:
+    # get dimensions of original image
```

Comment:
```suggestion
    input = input.to(dtype=torch.float32) if input.dtype == torch.float16 else input
    
```

Reply:
You also need to downcast it at the end.

You could also just try adding a `@pw_cast_for_opmath` decorator.